server/src/main/java/org/elasticsearch/search/aggregations/metrics/TDigestState.java

```diff
@@ -210,14 +210,19 @@ public static TDigestState read(CircuitBreaker breaker, StreamInput in) throws I
             breaker.addWithoutBreaking(-SHALLOW_SIZE);
             throw e;
         }
-        int n = in.readVInt();
-        if (size > 0) {
-            state.tdigest.reserve(size);
-        }
-        for (int i = 0; i < n; i++) {
-            state.add(in.readDouble(), in.readVLong());
+        try {
```

Comment:
I would prefer a try / finally block and use a boolean variable to control the outcome of the execution as we do in other places. See https://github.com/elastic/elasticsearch/blob/e2c314eec8bd7190a56f85c52342521491a5098e/server/src/main/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlus.java#L74

Reply:
Updated. Also, changed the internal catch to a finally too. merging both try-finally in one made the code a bit more complex, so left them separated

Reply:
Thanks you, that's looks good.